File: client/src/types/video.ts
export type VideoType = {
  id: string;
  name: string;
  original_filename: string;
  file_path: string;
  file_size: number;
  duration: number | null;
  width: number | null;
  height: number | null;
  format: string | null;
  codec: string | null;
  upload_date: string;
};

File: server/app/models/video.py
from sqlalchemy import Column, String, Float, Integer, DateTime
from datetime import datetime
from app.database import Base

class Video(Base):
    __tablename__ = "videos"

    id = Column(String, primary_key=True, index=True)
    name = Column(String, nullable=False)
    original_filename = Column(String, nullable=False)
    file_path = Column(String, nullable=False)
    file_size = Column(Integer, nullable=False)
    duration = Column(Float, nullable=True) 
    width = Column(Integer, nullable=True)
    height = Column(Integer, nullable=True)
    format = Column(String, nullable=True)
    codec = Column(String, nullable=True)
    upload_date = Column(DateTime, default=datetime.utcnow)